CMake/Dependencies.cmake

```diff
@@ -180,6 +180,7 @@ macro_log_feature(FREETYPE_FOUND "freetype" "Portable font engine" "http://www.f
 if (UNIX AND NOT APPLE AND NOT ANDROID AND NOT EMSCRIPTEN)
   find_package(X11)
   macro_log_feature(X11_FOUND "X11" "X Window system" "http://www.x.org" TRUE "" "")
+  add_compile_definitions(USE_X11)
```

Comment:
I added `add_compile_definitions(USE_X11)` because there are several examples of the same approach in the main `CMakeLists.txt` and its inclusions (with the older `add_definitions` command). Moreover, `ConfigureBuild.cmake` defines some configuration macros in generated header files like `OgreBuildSettings.h`.

In my opinion, letting a header file define the wrong `typedef` for `NativeDisplayType` and then patching potentially many type disagreements with explicit casts is worse and more difficult to maintain. Moreover, there may be other effects of the wrong detection of X11 apart from that definition, now or in the future.

The definition of `NativeDisplayType` is only loaded through [`OgreEGLSupport.h`](https://github.com/OGRECave/ogre/blob/464e87883075ebe71918a48527dc5ed149de7c73/RenderSystems/GLSupport/include/EGL/OgreEGLSupport.h#L35), and the X11-specific code (where casts would be needed) only includes `OgreEGLSupport.h` through `OgreX11EGLSupport.h`. Hence, I propose adding a `#define USE_X11` in `OgreX11EGLSupport.h` to limit the scope of the definition.

Reply:
I have pushed a new commit with the change proposed in the last paragraph. It works.